src/test/java/com/exasol/adapter/request/renderer/PushdownSqlRendererTest.java

```diff
@@ -0,0 +1,762 @@
+package com.exasol.adapter.request.renderer;
+
+import static com.exasol.adapter.metadata.DataType.createDecimal;
+import static com.exasol.adapter.metadata.DataType.createVarChar;
+import static com.exasol.adapter.metadata.DataType.ExaCharset.UTF8;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.file.*;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.json.*;
+
+import org.json.JSONException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+import com.exasol.adapter.metadata.ColumnMetadata;
+import com.exasol.adapter.metadata.TableMetadata;
+import com.exasol.adapter.request.parser.PushdownSqlParser;
+import com.exasol.adapter.sql.SqlNode;
+
+class PushdownSqlRendererTest {
```

Comment:
I have a feeling that the test is not isolated completely because it's deserializer and serializer together. You sure you don't care about a Java object between them?

Reply:
There is also another tests that tests the parser (deserializer). By that we know that the deserializer and the java object are correct. So we can use it here safely.

Reply:
Then a follow-up question: does it make sense to you to merge those two tests? They are both quite long and use exactly the same JSON files, so it's a lot of duplication

Reply:
I discussed this with @redcatbear since I also don't like the duplication. His opinion was that I should better duplicate it since its test code and there readability is more important than the DRY principle.

Reply:
If you would like to seperate the test setup would be in a different place then assertions.